hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/procedures/RunClusteringProcedure.scala

```diff
@@ -159,7 +150,12 @@ class RunClusteringProcedure extends BaseProcedure
     try {
       client = HoodieCLIUtils.createHoodieWriteClient(sparkSession, basePath, confs,
         tableName.asInstanceOf[Option[String]])
-
+      if (metaClient.getTableConfig.isMetadataTableAvailable) {
+        if (!confs.contains(HoodieLockConfig.LOCK_PROVIDER_CLASS_NAME.key)) {
+          confs = confs ++ HoodieCLIUtils.getLockOptions(basePath, metaClient.getBasePath.toUri.getScheme, client.getConfig.getCommonConfig.getProps())
+          logInfo("Auto config filesystem lock provider for metadata table")
```

Comment:
So the new logic become this when s3 is used and the lock provider config is missing: the auto generated lock provider config is empty here, but we still log a msg to indicate a fs lock provider is configured.

BTW, I check the `StorageSchemes` and only HDFS supports atomic creation of files, but this seems not correct, HDFS relies on `RENAME` to keep atomicity, while file creation is already atomic for most of the object storage like S3, GCS ..

Reply:
You're right. I'll fix the log.  StorageSchemes can open another issue for resolution.